CMakeLists.txt

```diff
@@ -236,6 +248,69 @@ else()
   message(STATUS "PETSc support disabled")
 endif()
 
+# Option: Ginkgo
+if (PRECICE_GinkgoParallelization)
+  find_package(Ginkgo 1.5 REQUIRED)
+  message(STATUS "Found Ginkgo ${GINKGO_PROJECT_VERSION} at ${GINKGO_INSTALL_PREFIX}")
+
+  # We aim for the same configuration as Ginkgo itself and check immediately for device compiler
+  # such that preciceCore inherits all relevant flags directly
+  # 1. Check if Ginkgo was built with CUDA and if CUDA is available on the system
+  if(${GINKGO_BUILD_CUDA})
+    # Necessary for the unified kernels and the qr decomposition
+    include(CheckLanguage)
+    check_language(CUDA)
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 17)
+    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+    set(CMAKE_CUDA_FLAGS "${GINKGO_CUDA_ARCH_FLAGS} --expt-relaxed-constexpr")
+
+    # Similar to Ginkgo, we don't let CMake deduce the architecture, but inherit the Ginkgo flags
+    if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.18)
+      set(CMAKE_CUDA_ARCHITECTURES OFF)
+    endif()
+    add_compile_definitions("PRECICE_WITH_CUDA")
```

Comment:
This should be added to the preciceCore target using `target_compile_definitions`.

Reply:
We need the compile definitions in our tests as well. This would duplicate the line then, right?

Reply:
We also need the deduced variables in the test configuration and in the testprecice target (cf [3a28c7c](https://github.com/precice/precice/pull/1630/commits/3a28c7cbe268026609beacb956e70523e7fbe093))